static/js/settings.js

```diff
@@ -119,6 +124,90 @@ var SettingsScreen = {
     });
   },
 
+  showAdapterSettings: function() {
+    this.adapterSettings.classList.remove('hidden');
```

Comment:
Moving this after the div creation would prevent a momentary flash of an empty list